src/test/regress/citus_tests/run_test.py

```diff
@@ -201,16 +201,39 @@ def default_base_schedule(test_schedule):
         shutil.copy2(
             os.path.join(regress_dir, dependencies.schedule), tmp_schedule_path
         )
+
+    def test_file_path(test_name):
+        test_suffix = "sql"
+        if dependencies.schedule == "base_isolation_schedule":
+            test_suffix = "spec"
+        test_file_dir = os.path.join(regress_dir, test_suffix)
+        test_file_name = f"{test_name}.{test_suffix}"
+        return os.path.join(test_file_dir, test_file_name)
+
+    # default worker count is 2, but we set it to higher value if we
+    # detect that any test requires more workers
+    worker_count = 2
+
+    def worker_count_for(test_name):
+        with open(test_file_path(test_name), "r") as testfile:
+            content = testfile.read()
+            worker_ports = re.findall(":worker_([0-9]+)_port", content)
+            ports = [int(worker_port) for worker_port in worker_ports]
+            return max(ports, default=2)
```

Comment:
While I think this is a nice trick. I think we should go for a simpler solution for now. The amount of tests that use more than 2 workers is currently none, and we expect at most 2 of such tests in the near future. This adds quite a bit of extra complexity to the `run_test.py` script for those very few cases. I think a simpler solution for now is adding a `worker_count` field to the `TestDeps` class and adding specific entry in the `deps` dict for these few tests.

Reply:
I reverted the commit to reference back if needed and  used `TestDeps` to configure worker count for now.